cabin-sorting/api/sortingData.js

```diff
@@ -0,0 +1,96 @@
+import path from "path";
+import fs from "fs";
+import { client, connectToDatabase } from "./mongodb";
+
+let validatedHackers;
+const validHackers = [];
+
+// Get all valid hacker data from the applicant_data database and convert to json contents
+async function grabFromDatabase() {
+  const applicantData = await connectToDatabase("applicant_data");
+
+  try {
+    // Query for hackers whose application status is submitted and who has a
+    // 'Post Acceptance Response' section
+    const query = {
+      applicationStatus: "Submitted",
+      postAcceptanceResponses: { $exists: true },
+      isAdmin: false,
+      rsvpStatus: "Confirmed",
+      "postAcceptanceResponses.swag": { $exists: false },
+      "postAcceptanceResponses.club": { $exists: true },
+    };
+
+    // Find all valid hackers that have answers to the cabin questions
+    const hackerDataCursor = applicantData
+      .find(query)
+      .project({ email: 1, postAcceptanceResponses: 1 });
+
+    // create list of hackers with their information/questions from the cursor
+    const hackerDocuments = await hackerDataCursor.toArray();
+
+    // parse and format each hacker in the list of hackers
+    hackerDocuments.map((item) => parseAndFormatHacker(item));
+
+    // get final filtered list of valid hackers
+    validatedHackers = validateHackers(validHackers);
+
+    // convert hacker list to json and output data to fromMongoDB.json
+    const hackerJson = JSON.stringify(validatedHackers);
+    const pathToWrite = path.resolve(
+      __dirname,
+      "../",
+      "data",
+      "json_outputs",
+      "fromMongoDB.json"
+    );
+    fs.writeFileSync(pathToWrite, hackerJson);
+  } finally {
+    // Ensures that the client will close when you finish/error
+    await client.close();
+  }
+  return validatedHackers;
+}
+
+// used for parsing hacker data into a uniform, correct, format
+function parseAndFormatHacker(item) {
+  // We are taking out the postAcceptanceResponses fields and putting them on
+  // same level as other fields (for easier access)
+  const postAcceptanceResponses = item.postAcceptanceResponses;
+  delete item.postAcceptanceResponses;
+  item = { ...item, ...postAcceptanceResponses };
+
+  // duplicate postAcceptanceResponse fields and rename fields to question<> : value
+  // so we can map these question answers to cabin answers
+  // (for cabin sorting in hackerSortingAlgo.ts)
+  const attributes = Object.keys(item);
+  for (let i = 0; i <= 9; i++) {
+    const j = attributes.length - 9 + i;
+    const attributevalue = item[attributes[j]];
+    // delete item[attributes[j]]
+    item[`question${i}`] = attributevalue;
+  }
+  // adding all hackers to list
+  validHackers.push(item);
+}
+
+// returns valid hacker list and handles errors if data doesn't exist or is empty
+function validateHackers(hackers) {
+  if (hackers === null) {
+    console.error(
```

Comment:
console.error => ret doesn't make sense to me? wouldn't it be console.warn => ret?

Reply:
yep, I agree that makes more sense!